### PR TITLE
loops_simple_programs_with_repitition_easy_q5

### PR DESCRIPTION
**Pull request by SanaBob**
loops_simple_programs_with_repitition_easy_q5

### CODE REVIEW

File: src/loops/simple_programs_with_repitition/easy/WordCount.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package loops.simple_programs_with_repitition.easy;
+
+/*
+Write a program that will ask the user for a sentence and
```
Comment (Hedgemon4) [inline src/loops/simple_programs_with_repitition/easy/WordCount.java:4]: Again, there is no sample code for the question, so the clause will mark them wrong, and we have not provided a correct formating for them. Code looks good though

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/loops/simple_programs_with_repitition/easy/WordCount.java (hunk 1 above)
Comment (Hedgemon4) [reply src/loops/simple_programs_with_repitition/easy/WordCount.java]: Nevermind, you should not need sample code for this one

**Review by Hedgemon4 — APPROVED** [review]

File: src/loops/simple_programs_with_repitition/easy/q5/WordCountTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+        return output;
+
+    }
+
+    public void runMain() {
+        provideInput(input);
+        WordCount.main(new String[0]);
+    }
+
```
Comment (SamamthaH) [inline src/loops/simple_programs_with_repitition/easy/q5/WordCountTest.java:26]: Again, you know the number of words you are providing for input. I don't believe you need the actual method.

File: src/loops/simple_programs_with_repitition/easy/q5/WordCountTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import global.variables.clauses.*;
+import global.BaseTest;
+
+public class WordCountTest extends BaseTest {
+
+    String input = "This is just for testing. Do not copy!";
+
+    public Clause[] testSentence() {
+
+        Clause[] output = new Clause[3];
```
Comment (Opeyem1a) [inline src/loops/simple_programs_with_repitition/easy/q5/WordCountTest.java:13]: Let's return the clause array the normal way unless we need to do it with indices

File: src/loops/simple_programs_with_repitition/easy/q5/WordCountTest.java (hunk 2 above)
Comment (Opeyem1a) [reply src/loops/simple_programs_with_repitition/easy/q5/WordCountTest.java:26]: I think this one is fine - eventually, you'll be providing a list of input strings and even if you store an array of their corresponding number of words, you'll still need a method to return the number of words for a specific input string

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: src/loops/simple_programs_with_repitition/easy/q5/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void testWithInput(String input) throws InvalidClauseException{
+        runWithInput(input);
+        assertEquals(Integer.parseInt(getItemByName("numberOutput")), countingWords(input), "Calculated value of output is incorrect");
+    }
+
+}
```
Comment (Opeyem1a) [inline src/loops/simple_programs_with_repitition/easy/q5/MainTest.java:56]: EOF newline

File: src/loops/simple_programs_with_repitition/easy/q5/MainTest.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            new StringLiteral(" words.")
+        };
+
+    }
+
+    public void runMain() {
+        WordCount.main(new String[0]);
+    }
+
+    public static int countingWords(String message) {
```
Comment (Opeyem1a) [inline src/loops/simple_programs_with_repitition/easy/q5/MainTest.java:36]: Don't rely this heavily on helper methods, we should hardcode input/output in the `inputProvider` in these cases

File: src/loops/simple_programs_with_repitition/easy/q5/MainTest.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        message.toLowerCase();
+        int count = 0;
+        for (int i = 0; i < message.length(); i++)
+            if (message.charAt(i) == ' ')
+                count++;
+        return count + 1;
+    }
+
+    static Stream<Arguments> inputProvider(){
+        return Stream.of(Arguments.of("This is just for testing. Do not copy!"), Arguments.of("This is second test"), Arguments.of(" ")); // Doesn't work with empty string
```
Comment (Opeyem1a) [inline src/loops/simple_programs_with_repitition/easy/q5/MainTest.java:46]: If it's not supposed to work with an empty string, that is concern to raise in the question report channel @SanaBob

Comment (Opeyem1a) [reply src/loops/simple_programs_with_repitition/easy/q5/MainTest.java:46]: I also think we should have a few more cases here

File: src/loops/simple_programs_with_repitition/easy/q5/MainTest.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void testWithInput(String input) throws InvalidClauseException{
+        runWithInput(input);
+        assertEquals(Integer.parseInt(getItemByName("numberOutput")), countingWords(input), "Calculated value of output is incorrect");
```
Comment (Opeyem1a) [inline src/loops/simple_programs_with_repitition/easy/q5/MainTest.java:53]: What is being "calculated" in the output? This error message needs updating

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: src/global/BaseTest.java [hunk 8]
```diff
@@ -206,4 +206,4 @@ public void restoreSystemInputOutput() {
         System.setIn(systemIn);
         System.setOut(systemOut);
     }
-}
+}
```
Comment (Opeyem1a) [inline src/global/BaseTest.java:209]: Avoid changes to the framework files themselves

File: src/loops/simple_programs_with_repitition/easy/q5/MainTest.java (hunk 5 above)
Comment (Opeyem1a) [reply src/loops/simple_programs_with_repitition/easy/q5/MainTest.java:36]: You should remove this method once it's not used

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: src/global/BaseTest.java (hunk 8 above)
Comment (SanaBob) [reply src/global/BaseTest.java:209]: It happenned upon merging main, I tried everything but it still views it as a change

Comment (Opeyem1a) [inline src/global/BaseTest.java:209]: Pls add the EOF newline back

File: src/loops/simple_programs_with_repitition/easy/q5/MainTest.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+    static Stream<Arguments> inputProvider(){
+        return Stream.of(Arguments.of("This is just for testing. Do not copy!", 8), Arguments.of("This is second test", 4), Arguments.of("Another hardcoded test is over here.", 6), Arguments.of(" ", 2)); // Doesn't work with empty string
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void testWithInput(String input, int output) throws InvalidClauseException{
+        runWithInput(input);
+        assertEquals(Integer.parseInt(getItemByName("numberOutput")), output, "Total number of words is incorrect");
```
Comment (Opeyem1a) [inline src/loops/simple_programs_with_repitition/easy/q5/MainTest.java:44]: Add a period `.` at the end of this error message

File: src/loops/simple_programs_with_repitition/easy/q5/MainTest.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        };
+
+    }
+
+    public void runMain() {
+        WordCount.main(new String[0]);
+    }
+
+    static Stream<Arguments> inputProvider(){
+        return Stream.of(Arguments.of("This is just for testing. Do not copy!", 8), Arguments.of("This is second test", 4), Arguments.of("Another hardcoded test is over here.", 6), Arguments.of(" ", 2)); // Doesn't work with empty string
```
Comment (Opeyem1a) [inline src/loops/simple_programs_with_repitition/easy/q5/MainTest.java:37]: While you're here, put each `Arguments.of()` on a new line for readability

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

**Review by Opeyem1a — APPROVED** [review]

**Pull request merged**